### PR TITLE
Defect/566 tenant aware school year

### PR DESCRIPTION
**Pull request by esotrope**
affected queries:

1. examPermissionTest.findAllByStudentId
2. exam.findAllIabsByStudentIdsAndQueryParamsUnknownPermissions
3. exam.findAllIabsByStudentIdsAndQueryParamsGroupPermissions
4. exam.findAllIcasOrSummativesByStudentIdsAndQueryParamsUnknownPermissions
5. exam.findAllIcasOrSummativesByStudentIdsAndQueryParamsGroupPermissions
6. export.exams
7. student.findByExamQueryParamsUnknownPermissions
8. student.findByExamQueryParamsGroupPermissions
9. exam.findAllByStudentId
10. exam.existsForStudentId
11. examItem.findAllForExam
12. group.assessment.findAll
13. group.assessment.findLatestExam
14. group.exam.findAllForAssessment
15. group.examWithTargetScore.findAllForAssessment
16. group.examItem.findAllExamItemScoresForAssessment
17. schoolgrade.assessment.findAll
18. schoolgrade.assessment.findLatestExam
19. schoolgrade.exam.findAllForAssessment
20. schoolgrade.examItem.findAllExamItemScoresForAssessment
21. assessment.grade.findAllForSchool
22. addons.districtEmbargo
23. addons.stateEmbargo

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/AbstractReportRepository.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
         <T extends AggregateRow, R extends AggregateRowReportResult<T>, Q extends OrganizationTypeQuery, B extends AggregateRowReportResult.Builder<T, B, R>>
         implements ReportRepository<T, R, Q> {
 
-    private final ReportingSystemProperties reportingSystemProperties;
+    protected final ReportingSystemProperties reportingSystemProperties;
```
Comment (esotrope) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/impl/AbstractReportRepository.java]: saved me from having to pass this into the getQuery methods

File: aggregate-service/src/main/resources/claim-report-template.sql.yml [hunk 2]
```diff
@@ -105,15 +105,15 @@ sqlbuilder:
             districtEmbargo:
               clauses:
                 where: >-
-                    e.school_year < ${reporting.school-year}
+                    e.school_year < :system_school_year
```
Comment (esotrope) [inline aggregate-service/src/main/resources/claim-report-template.sql.yml]: changed placeholder to named param

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/jdbc/SecurityParameterProvider.java [hunk 3]
```diff
@@ -35,7 +33,8 @@
     private final ReportingSystemProperties reportingSystemProperties;
 
     @Autowired
-    public SecurityParameterProvider(@Qualifier("reportingSystemPropertiesResolver") final ReportingSystemProperties reportingSystemProperties) {
+    public SecurityParameterProvider(
+            @Qualifier("reportingSystemPropertiesResolver") final ReportingSystemProperties reportingSystemProperties) {
```
Comment (esotrope) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/jdbc/SecurityParameterProvider.java]: at first i went through painstakingly making only the queries that needed the param have it however i noticed that in _all_ instances they were secured sql calls and had the user's security parameters so i found it much less code-invasive to make it a common security parameter. This does mean that `system_school_year` will be a reserved name parameter of sorts though

File: reporting-service/src/main/resources/application.sql.yml [hunk 4]
```diff
@@ -87,50 +87,50 @@ sql:
       LIMIT 1
 
   examItem:
-      findAllForExam: >-
-        ${sql.reporting.snippet.selectFromExamItem}
-        WHERE e.id = :exam_id
-        AND a.type_id != 3
-        AND ${sql.reporting.snippet.studentExamPermissionWithEmbargo}
+    findAllForExam: >-
```
Comment (esotrope) [inline reporting-service/src/main/resources/application.sql.yml]: indentation

File: aggregate-service/src/test/java/org/opentestsystem/rdw/olap/sqlbuilder/impl/query/ClaimReportQueryTemplateProviderRST.java [hunk 5]
```diff
@@ -34,6 +36,11 @@
         QueryProvider.class
 })
 public class ClaimReportQueryTemplateProviderRST extends RepositoryBackedIT {
```
Comment (esotrope) [inline aggregate-service/src/test/java/org/opentestsystem/rdw/olap/sqlbuilder/impl/query/ClaimReportQueryTemplateProviderRST.java]: @malaffoon  ran all tests but MT and RST.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/impl/query/ClaimSqlQueryWithReportTemplateProvider.java [hunk 6]
```diff
@@ -29,12 +31,14 @@
...
-                                                   final QueryProviderRepositoryHelper repositoryHelper,
-                                                   final PermissionSource permissionSource) {
-        super(new ClaimReportTemplateProvider(state, repositoryHelper), organizationPartitionSize, queryProvider, repositoryHelper, permissionSource);
+    public ClaimSqlQueryWithReportTemplateProvider(
+            final Organization state,
+            final int organizationPartitionSize,
+            final QueryProvider queryProvider,
+            final QueryProviderRepositoryHelper repositoryHelper,
+            final PermissionSource permissionSource,
+            @Qualifier("reportingSystemPropertiesResolver") final ReportingSystemProperties reportingSystemProperties) {
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/impl/query/ClaimSqlQueryWithReportTemplateProvider.java]: was it not finding it without the qualifier?

Comment (esotrope) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/impl/query/ClaimSqlQueryWithReportTemplateProvider.java]: there are two beans mapped to `ReportingSystemProperties` one is this one that resolves using the active tenant the other is...??? not sure @ftdholbrook ? : J

**Pull request merged**